### DISCUSSION TITLE
Manual offline DBA Dash update/upgrade

### DISCUSSION THREAD

**Discussion by BrubiSQL** [trimble-oss/dba-dash#130] [Show and tell]
Considering it's not written in the documentation, here are the steps for manual (offline) tool upgrade

1. Download the latest version (all of it) and unzip the package
2. Stop the DBA Dash service
3. Make a backup of DBA Dash database
4. Make a backup of ServiceConfig.json file
5. Open up the folder where you initially placed your service exe and DLL's - overwrite the folder with new version (earlier downloaded files), be careful that you don't overwrite ServiceConfig.json file
6. Start the DBADashServiceConfigTool.exe and click on the "Deploy/Update Database" button 
-- On this step you can click on the "Generate", so you can check Database changes
-- To deploy the changes on the DBA Dash database click on the "Deploy"
7. After you have finished all the previous steps, you can start the DBA Dash service

Aaaaand that's it!
Your DBA Dash is now manually updated :)
Also, be mindful of the time of day/week you are doing this, you don't want DBA Dash to start scanning DB Schema after you re-start the service - or you could simply turn off this feature, and rely on your schedule to get your schema scan :)

Best regard,
B.
:)

**Comment by DavidWiseman** [comment]
Thank you.  Most of the collections are set to run on service start.  This provides a method to trigger collection ahead of it's schedule by restarting the service.  
If you didn't want a collection to trigger on service start it's possible to disable it.  In the service configuration tool open the Options tab.  Click the Configure Schedule button.
![image](https://user-images.githubusercontent.com/33157668/162395322-fc6820df-5926-4eec-9ee6-6ed3ec87a637.png)
If default is checked the schedule is controlled by the application defaults - it's possible these will get updated with new versions of the app and should be suitable for most users.  To override, uncheck default.  You can then uncheck run on service start and enter a custom cron expression for the schedule if required.
Schema snapshots could take a while to run depending on the number of objects in the database and the number of databases.  The process is single threaded.  It could be done faster with some multi-threading but the main goal is to limit the impact on the monitored instance.  Ideally it still runs out of hours.
Schedules can also be configured on an instance by instance basis by clicking the "Schedule" link in the existing connections grid (Source tab).

Hope this helps.
